### PR TITLE
feat: adds the possibility to rename a clan

### PR DESCRIPTION
**Pull request by Tomut0**
🐸 ?

### CODE REVIEW

File: src/main/resources/plugin.yml [hunk 1]
```diff
@@ -140,6 +140,8 @@ permissions:
     description: Can set his clan's banner
   simpleclans.leader.discord.create:
     description: Can create discord channel for his clan
+  simpleclans.leader.rename:
```
Comment (RoinujNosde) [inline src/main/resources/plugin.yml:143]: `false` by default

File: src/main/resources/messages.properties [hunk 2]
```diff
@@ -228,6 +229,7 @@ command.description.rank.permissions.available=Lists the available rank permissi
 command.description.rank.permissions.rank=Lists a rank''s permissions
 command.description.rank.permissions.add=Adds a permission to a rank
 command.description.rank.permissions.remove=Removes a permission from a rank
+command.description.rename=Renames a clan
```
Comment (RoinujNosde) [inline src/main/resources/messages.properties]: There should also be the player equivalent: "Renames **your** clan"

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/commands/clan/LeaderCommands.java [hunk 3]
```diff
@@ -187,6 +184,22 @@ public void untrust(Player player, Clan clan, @Conditions("same_clan") @Name("me
...
     }
 
+    @Subcommand("%rename")
+    @CommandPermission("simpleclans.leader.rename")
+    @CommandCompletion("@nothing")
+    @Description("{@@command.description.rename}")
+    public void rename(ClanPlayer cp, Clan clan, @Name("name") String clanName) {
+        if (clan.getLeaders().size() != 1) {
+            requestManager.requestAllLeaders(cp, ClanRequest.RENAME, "asking.to.rename", clanName);
+            return;
```
Comment (RoinujNosde) [inline src/main/java/net/sacredlabyrinth/phaed/simpleclans/commands/clan/LeaderCommands.java:195]: Send message "... vote has been requested"?

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/RequestManager.java [hunk 4]
```diff
@@ -52,37 +52,53 @@ public void addDemoteRequest(ClanPlayer requester, String demotedName, Clan clan
...
+     *     <li>Disband request can be asked from all leaders</li>
+     *     <li>Rename request can be asked from all leaders</li>
+     *     <li>Promote request can be asked from all leaders</li>
+     * </ul>
+     *
+     * <p>
+     * Examples of incompatible requests:
+     * </p>
+     * <ul>
+     *      <li>Demote request can be asked from all leaders,
```
Comment (RoinujNosde) [inline src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/RequestManager.java:76]: What about throwing an InvalidArgumentException if ClanRequest == DEMOTE or INVITE?

**Review by RoinujNosde — CHANGES_REQUESTED** [review]

File: src/main/resources/messages.properties (hunk 2 above)
Comment (Tomut0) [reply src/main/resources/messages.properties]: Description is used in staff command as well. So, not only your clan, right? 👀

File: src/main/resources/plugin.yml (hunk 1 above)
Comment (Tomut0) [reply src/main/resources/plugin.yml:143]: Why `false`? Leaders shouldn't have the possibility to rename their clan by default?

File: src/main/resources/messages.properties (hunk 2 above)
Comment (Tomut0) [reply src/main/resources/messages.properties]: изображение
as example, demote command:

File: src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/RequestManager.java [hunk 5]
```diff
@@ -52,37 +52,53 @@ public void addDemoteRequest(ClanPlayer requester, String demotedName, Clan clan
...
+     *      <li>Invite request has to ask someone <b>outside</b> of leaders clan</li>
+     * </ul>
+     *
+     * @param requester the clan player, who sent the request
+     * @param request   the type of request, see: {@link ClanRequest}
+     * @param target    the target which will be used in request processing
+     * @param key       the language key that would be translated and send the message to all leaders
+     * @param args      the language objects, requires in some language strings.
+     */
+    public void requestAllLeaders(@NotNull ClanPlayer requester, @NotNull ClanRequest request,
```
[the remark this answers is not in the document]
Comment (Tomut0) [reply src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/RequestManager.java:88]: No refactor for you, codeclimate 👺 🔫

File: src/main/resources/plugin.yml (hunk 1 above)
Comment (RoinujNosde) [reply src/main/resources/plugin.yml:143]: Yeah, some servers might not want that.

File: src/main/resources/messages.properties (hunk 2 above)
Comment (RoinujNosde) [reply src/main/resources/messages.properties]: command.description.home.tp
command.description.mod.home.tp

**Review by RoinujNosde — APPROVED** [review]

**Comment by RoinujNosde** [comment]
Thanks! ;)

**Pull request merged**